lib/karto_sdk/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.5)
 project(karto_sdk)
 
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
This doesn't compile, change to C++17 in the make cmakelists file, this won't actually be set here because `CMAKE_CXX_STANDARD` is set in the main CMakelists file as C++14

Reply:
I had to revert the noetic one too for the same reasons

Reply:
```
In file included from /home/steve/Documents/nav2_ws/src/slam_toolbox-1/lib/karto_sdk/src/Karto.cpp:27:
/home/steve/Documents/nav2_ws/src/slam_toolbox-1/lib/karto_sdk/include/karto_sdk/Karto.h:5438:16: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
 5438 |   mutable std::shared_mutex m_Lock;
      |                ^~~~~~~~~~~~
/home/steve/Documents/nav2_ws/src/slam_toolbox-1/lib/karto_sdk/include/karto_sdk/Karto.h:5438:11: note: ‘std::shared_mutex’ is only available from C++17 onwards
 5438 |   mutable std::shared_mutex m_Lock;
      |           ^~~

```